library/src/options.cxx

```diff
@@ -119,7 +119,11 @@ F3D_DECL_TYPE(int);
 F3D_DECL_TYPE(double);
 F3D_DECL_TYPE(f3d::ratio_t);
 F3D_DECL_TYPE(std::string);
+F3D_DECL_TYPE(std::vector<bool>);
```

Comment:
Actually we can factorize the macro to declare both vector/single pair with one call.
```cpp
#define F3D_DECL_TYPE_INTERNAL(TYPE) template F3D_EXPORT TYPE options::parse<TYPE>(const std::string& str)
#define F3D_DECL_TYPE(TYPE) \
F3D_DECL_TYPE_INTERNAL(TYPE); \
F3D_DECL_TYPE_INTERNAL(std::vector<TYPE>)
```
There are probably missing parentheses but you got the point.